mountpoint-s3-fs/src/prefetch/part_stream.rs

```diff
@@ -161,30 +156,62 @@ impl Debug for RequestRange {
     }
 }
 
+/// Type-erased [ObjectPartStream].
+#[derive(Clone)]
+pub struct PartStream<Client> {
+    inner: Arc<dyn ObjectPartStream<Client> + Send + Sync + 'static>,
+}
+
+impl<Client> PartStream<Client>
+where
+    Client: ObjectClient + Clone + Send + Sync + 'static,
+{
+    pub fn new<Stream>(part_stream: Stream) -> Self
+    where
+        Stream: ObjectPartStream<Client> + Send + Sync + 'static,
+    {
+        Self {
+            inner: Arc::new(part_stream),
+        }
+    }
+
+    pub fn spawn_get_object_request(&self, config: RequestTaskConfig) -> RequestTask<Client> {
+        self.inner.spawn_get_object_request(config)
+    }
+
+    pub fn client(&self) -> &Client {
+        self.inner.client()
+    }
+}
+
+impl<Client> Debug for PartStream<Client> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PartStream").finish()
+    }
+}
```

Comment:
nit: prefer using [`finish_non_exhaustive`](https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html#method.finish_non_exhaustive) when we skip some fields?

```suggestion
impl<Client> Debug for PartStream<Client> {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.debug_struct("PartStream").finish_non_exhaustive()
    }
}
```